### PR TITLE
Updates to session keys

### PR DESCRIPTION
**Pull request by joepetrowski**

### CODE REVIEW

**Comment by JoshOrndorff** [comment]
I've always been curious how much of this is only relevant when pallet_sessions is installed, and how much is still relevant even without that pallet.

File: docs/conceptual/cryptography/session-keys.md [hunk 1]
```diff
@@ -4,25 +4,34 @@ title: Session Keys
...
+You can declare any number Session keys. For example, the default Substrate node uses four. Other chains could have more or fewer depending on what operations the chain expects its validators to perform.
 
-You can declare any number Session keys. For example, the default Substrate node uses three. Other chains could have more or fewer depending on what operations the chain expects its validators to perform.
+In practice, validators amalgamate all of the session public keys into a certificate by signing the set of public keys with a "Controller" account and submitting a transaction. This on-chain certificate links a validator node with an account that holds funds. As such, that account can be credited with rewards or slashed based on the node's behavior.
+
+The runtime declares what session keys will be included in a certificate (`runtime/src/lib.rs`):
 
 ```rust
 pub struct SessionKeys {
```
Comment (bkchr) [inline docs/conceptual/cryptography/session-keys.md]: This needs to be wrapped into [impl_opaque_keys!](https://github.com/paritytech/substrate/blob/master/primitives/runtime/src/traits.rs#L1142)

File: docs/conceptual/cryptography/session-keys.md [hunk 2]
```diff
@@ -4,25 +4,34 @@ title: Session Keys
...
+The default Substrate node implements Session keys in the [Session pallet](https://substrate.dev/rustdocs/master/pallet_session/).
 
 ## Generation and Use
 
-Session keys are hot keys that must be kept online. They are not meant to be used as account keys. If one of the Session keys is compromised, the attacker could commit slashable behavior. Session keys may be changed regularly (e.g. every session) via [RPC](https://substrate.dev/rustdocs/master/sc_rpc/author/trait.AuthorApi.html#tymethod.rotate_keys) for increased security.
+Session keys are hot keys that must be kept online. The individual keys should **not** be used to control funds, and the keys should be treated as a single certificate rather than as individual keys. All the logic for handling them is in the Substrate client, primitives, and Session pallet. If one of the Session keys is compromised, the attacker could commit slashable behavior. Session keys may be changed regularly (e.g. every session) via [RPC](https://substrate.dev/rustdocs/master/sc_rpc/author/trait.AuthorApi.html#tymethod.rotate_keys) for increased security.
```
Comment (bkchr) [inline docs/conceptual/cryptography/session-keys.md]: "as a single certificate"

I always find this confusing to call this stuff certificates. I'm also not sure what exactly you mean by this.

Comment (bkchr) [inline docs/conceptual/cryptography/session-keys.md]: For the last sentence. It is not enough to call `rotate_keys`, you also need to register them on chain.

File: docs/conceptual/cryptography/session-keys.md [hunk 3]
```diff
@@ -4,25 +4,34 @@ title: Session Keys
...
 
-To create a Session key, validator operators must attest that a key acts on behalf of their Stash account (stake) and nominators. To do so, they create a certificate by signing the key with their Controller key. Then, they inform the chain that this key represents their Controller key by publishing the Session certificate in a transaction on the chain.
+You can declare any number Session keys. For example, the default Substrate node uses four. Other chains could have more or fewer depending on what operations the chain expects its validators to perform.
 
-You can declare any number Session keys. For example, the default Substrate node uses three. Other chains could have more or fewer depending on what operations the chain expects its validators to perform.
+In practice, validators amalgamate all of the session public keys into a certificate by signing the set of public keys with a "Controller" account and submitting a transaction. This on-chain certificate links a validator node with an account that holds funds. As such, that account can be credited with rewards or slashed based on the node's behavior.
```
Comment (bkchr) [inline docs/conceptual/cryptography/session-keys.md]: "certificate" again.

File: docs/conceptual/cryptography/session-keys.md [hunk 4]
```diff
@@ -28,10 +37,21 @@ mod app {
 }
 ```
 
-> **Note:** This code is just an example of the Substrate node at the [time of writing](https://github.com/paritytech/substrate/tree/9fa8589d9b8cfe8716e9e4c48f9e3f238c1e502f). Refer to the runtime for the most up-to-date implementation.
+> **Note:** This code is just an example of the Substrate node at the
```
Comment (JoshOrndorff) [inline docs/conceptual/cryptography/session-keys.md]:
```suggestion
> This code is just an example of the Substrate node at the
```

My stylistic preference. It wouldn't be written in that yellow box if we didn't want readers to note it.

Comment (JoshOrndorff) [reply docs/conceptual/cryptography/session-keys.md]: Up to you though, Joe. I'm fine either way.

Comment (joepetrowski) [reply docs/conceptual/cryptography/session-keys.md]: I love cutting extra words!

**Pull request merged**